### ISSUE TITLE
Fix warning regarding incompatible function pointer types

### ISSUE THREAD

**Issue by wdeconinck** [FESOM/fesom2#512]
The fesom refactoring branch seems to have a problem with parms: I have following warnings
```c++
lib/parms/src/DDPQ/arms2.c:233:3: warning: incompatible function pointer types initializing 'int (*)(parms_Operator, void **)' (aka 'int (*)(struct parms_Operator_ *, void **)') with an expression of type 'int (parms_Operator, double *, double *)' (aka 'int (struct parms_Operator_ *, double *, double *)') [-Wincompatible-function-pointer-types]
  parms_arms_ascend_vcsr,
  ^~~~~~~~~~~~~~~~~~~~~~
lib/parms/src/DDPQ/arms2.c:234:3: warning: incompatible function pointer types initializing 'int (*)(parms_Operator, double *, double *)' (aka 'int (*)(struct parms_Operator_ *, double *, double *)') with an expression of type 'int (parms_Operator)' (aka 'int (struct parms_Operator_ *)') [-Wincompatible-function-pointer-types]
  parms_arms_getssize_vcsr,
  ^~~~~~~~~~~~~~~~~~~~~~~~
lib/parms/src/DDPQ/arms2.c:235:3: warning: incompatible function pointer types initializing 'int (*)(parms_Operator)' (aka 'int (*)(struct parms_Operator_ *)') with an expression of type 'void (parms_Operator, int *, int *)' (aka 'void (struct parms_Operator_ *, int *, int *)') [-Wincompatible-function-pointer-types]
  parms_arms_nnz,
  ^~~~~~~~~~~~~~
lib/parms/src/DDPQ/arms2.c:236:3: warning: incompatible function pointer types initializing 'void (*)(parms_Operator, int *, int *)' (aka 'void (*)(struct parms_Operator_ *, int *, int *)') with an expression of type 'int (parms_Operator *)' (aka 'int (struct parms_Operator_ **)') [-Wincompatible-function-pointer-types]
  arms_free_vcsr,
  ^~~~~~~~~~~~~~
lib/parms/src/DDPQ/arms2.c:237:3: warning: incompatible function pointer types initializing 'int (*)(parms_Operator *)' (aka 'int (*)(struct parms_Operator_ **)') with an expression of type 'int (parms_Operator, parms_Viewer)' (aka 'int (struct parms_Operator_ *, struct parms_Viewer_ *)') [-Wincompatible-function-pointer-types]
  arms_view_vcsr
  ^~~~~~~~~~~~~~
```
This is the struct which is being assigned is in `parms_opt_impl.h`:
```c++
typedef struct parms_Operator_ops{
 int  (*apply)(parms_Operator self, FLOAT *y, FLOAT *x);
 int  (*lsol)(parms_Operator self, FLOAT *y, FLOAT *x);
 int  (*invs)(parms_Operator self, FLOAT *y, FLOAT *x);
 int  (*getu)(parms_Operator self, void **mat);
 int  (*ascend)(parms_Operator self, FLOAT *y, FLOAT *x);
 int  (*getssize)(parms_Operator self);
 void (*getnnz)(parms_Operator self, int *nnz_mat, int
		       *nnz_pc);
 int  (*operator_free)(parms_Operator *self);
 int  (*operator_view)(parms_Operator self, parms_Viewer v); 
} *parms_Operator_ops; 
```
and gets created in `arms2.c`. I have added in comment the name of the function-ptr in the struct getting assigned.
```c++
static struct parms_Operator_ops parms_arms_sol_vptr = {
  parms_arms_sol_vcsr,        // apply
  parms_arms_lsol_vcsr,       // lsol
  parms_arms_invs_vcsr,       // invs
  parms_arms_ascend_vcsr,     // getu --> MISMATCH
  parms_arms_getssize_vcsr,   // ascend --> MISMATCH
  parms_arms_nnz,             // getssize --> MISMATCH
  arms_free_vcsr,             // getnnz --> MISMATCH
  arms_view_vcsr              // operator_free --> MISMATCH
                              // operator_view --> UNASSIGNED
};
```
Everything is shifted started from `ascend`, and I don't know how to fix to set the `getu` function correctly.

What is more correct but still requires fixes is by setting `getu` to NULL:
```c++
static struct parms_Operator_ops parms_arms_sol_vptr = {
  parms_arms_sol_vcsr,        // apply
  parms_arms_lsol_vcsr,       // lsol
  parms_arms_invs_vcsr,       // invs
  NULL,                       // getu --> UNASSIGNED!!!
  parms_arms_ascend_vcsr,     // ascend
  parms_arms_getssize_vcsr,   // getssize
  parms_arms_nnz,             // getnnz
  arms_free_vcsr,             // operator_free
  arms_view_vcsr              // operator_view
};
```

**Q: Why was this not discovered? Is this a dead code path?**

**Comment by wdeconinck** [comment]
@dsidoren, note that commit 74f796e merged with PR #523 only hides the warning, but does not fix the bug!

**Comment by JanStreffing** [comment]
Parms was removed in favor of Sergey solver. https://github.com/FESOM/fesom2/pull/597
